.circleci/config.yml

```diff
@@ -1869,6 +1869,52 @@ jobs:
           paths:
             - test-unit-harness-gpu-pycov
 
+  test-unit-harness-pytorch2-gpu:
+    docker:
+      - image: determinedai/environments:cuda-11.8-pytorch-2.0-gpu-8b7ced7
+    resource_class: determined-ai/container-runner-gpu
+    steps:
+      - run: mkdir -p ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
+      - checkout
+      - skip-if-docs-only
+      - skip-if-github-only
+      - skip-if-webui-only
+      - run: pip install mypy pytest coverage
+      - install-codecov
+      - run: echo 'export PATH=$PATH:$HOME/.local/bin' >> $BASH_ENV
+      - run: echo 'export PATH=$PATH:/usr/local/nvidia/bin' >> $BASH_ENV
+      - run: echo 'export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/local/nvidia/lib64' >> $BASH_ENV
+      - run: COVERAGE_FILE=/root/project/test-unit-harness-pytorch2-gpu-pycov make -C harness test-pytorch-gpu
+      - run: coverage xml -i --data-file=./test-unit-harness-pytorch2-gpu-pycov
+      - run: codecov -v -t $CODECOV_TOKEN -F harness
+      - persist_to_workspace:
+          root: .
+          paths:
+            - test-unit-harness-pytorch2-gpu-pycov
+
+  test-unit-harness-pytorch2-cpu:
+    docker:
+      - image: determinedai/environments:py-3.10-pytorch-2.0-cpu-8b7ced7
+    steps:
+      - run: mkdir -p ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
+      - checkout
+      - skip-if-docs-only
+      - skip-if-github-only
+      - skip-if-webui-only
+      - run: pip install mypy pytest coverage
+      - install-codecov
+      - run: echo 'export PATH=$PATH:$HOME/.local/bin' >> $BASH_ENV
+      - run: echo 'export PATH=$PATH:/usr/local/nvidia/bin' >> $BASH_ENV
+      - run: echo 'export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/local/nvidia/lib64' >> $BASH_ENV
+      - run: COVERAGE_FILE=/root/project/test-unit-harness-pytorch2-cpu-pycov make -C harness test-pytorch-cpu
+      - run: coverage xml -i --data-file=./test-unit-harness-pytorch2-cpu-pycov
+      - run: codecov -v -t $CODECOV_TOKEN -F harness
```

Comment:
can we not repeat these somewhat arcane steps every time, and instead have a `test-unit-harness` job parametetrized by
- docker image
- test name
- persisted coverage file path

Reply:
I condensed these a little bit just for good measure here, but not every one of these jobs is the same. Some cpu jobs use docker, some use the old venv structure, GPU jobs have an extra library update step. A redesign is probably in order, but I'd make that a separate PR.